src/parser/content-parser.php

```diff
@@ -32,9 +34,11 @@ public function __construct( $block_registry = null ) {
 	 *
 	 * @return array|WP_Error
 	 */
-	public function parse( $post_content, $post_id = null ) {
+	public function parse( $post_content, $post_id = null, $blocks_to_skip = [], $blocks_to_include = [] ) {
 		$this->post_id  = $post_id;
 		$this->warnings = [];
+		$this->blocks_to_include = $blocks_to_include;
```

Comment:
What do you think about adding a code filter here as well?

```php
$blocks_to_include = apply_filters( 'vip_block_data_api__include_blocks', $blocks_to_include );
```

This would give another useful way to define the include/exclude list in code, e.g.

```php
add_filter( 'vip_block_data_api__include_blocks', 'setup_included_blocks' );

function setup_included_blocks( $blocks_to_include ) {
    // Only allow heading, paragraph, and my-custom-block in API results
    return [
        'core/heading',
        'core/paragraph',
        'wpvip/my-custom-block',
    ];
}
```

I think this approach would be easier if there's a big list of included/excluded blocks and a GET request is getting unwieldy. The customer can choose if the client or server decides the blocks, which is nice.